app/models/filters/filter_base.rb

```diff
@@ -536,6 +554,16 @@ def effective_project_ids_from_project_groups
         pluck(p_t[:id].as('project_id'))
     end
 
+    def effective_project_ids_from_secondary_project_groups
+      projects = secondary_project_group_ids.reject(&:blank?).map(&:to_i)
+      return [] if projects.empty?
+
+      @effective_project_ids_from_secondary_project_groups ||= GrdaWarehouse::ProjectGroup.joins(:projects).
+        merge(GrdaWarehouse::ProjectGroup.viewable_by(user)).
+        where(id: projects).
```

Comment:
projects is mis-named here, these are actually project_group_ids, and will need to approach getting the project IDs differently.